database/data/mubi/mubi_all_festival_films_to_csv.py

```diff
@@ -0,0 +1,111 @@
+
+        self._init_csv_file()
+
+        try:
+            async with AsyncBrowserSession() as session:
+                for year in range(self.start_year, self.end_year + 1):
+                    for page_num in self.page_range:
+                        pair = (year, page_num)
+                        if pair in scraped_pairs:
+                            print(f"⏩ Skipping already scraped year={year}, page={page_num}")
+                            continue
+
+                        await self._scrape_and_append(session, year, page_num)
+
+                        request_count += 1
+                        if request_count >= max_requests_per_session:
+                            print(f"🔁 Restarting browser session after {request_count} requests.")
+                            await session.__aexit__(None, None, None)
+                            session = await AsyncBrowserSession().__aenter__()  # restart session
+                            request_count = 0
+
+                        delay = random.uniform(3, 5)
+                        print(f"⏳ Waiting {delay:.2f} seconds...")
+                        await asyncio.sleep(delay)
+
+        except PlaywrightError as e:
+            print(f"🔥 Browser session crashed: {e}")
+        except Exception as e:
+            print(f"💥 Unexpected error: {e}")
+
+    async def _scrape_and_append(self, session, year: int, page_num: int):
+        filters = {
+            "festival": self.festival,
+            "year": str(year),
+            "page_num": str(page_num)
+        }
+
+        url = self.scraper.FESTIVAL_EDITION_ALL_FILMS_URL.format(**filters)
+
+        try:
+            html = await session.fetch_html(url)
+            films = self.scraper.extract_festival_edition_all_films(html)
+
+            if not films:
+                print(f"❌ No films found at {year} page {page_num}.")
+                return
+            for film in films:
+                film["festival"] = self.festival
+                film["year"] = year
+                film["page_num"] = page_num
+
+            with open(self.csv_path, mode="a", newline="", encoding="utf-8") as f:
+                writer = csv.DictWriter(f, fieldnames=self.fieldnames)
+                writer.writerows(films)
+
+            print(f"✅ Saved {len(films)} films from {self.festival} {year}, page {page_num}")
+
+        except PlaywrightError as e:
+            print(f"⚠️ Session error at {year} page {page_num}: {e}")
+        except Exception as e:
+            print(f"❌ Failed at {year} page {page_num}: {e}")
+
+    def _init_csv_file(self):
+        if not self.csv_path.exists():
+            with open(self.csv_path, mode="w", newline="", encoding="utf-8") as f:
+                writer = csv.DictWriter(f, fieldnames=self.fieldnames)
+                writer.writeheader()
+
+    def _load_existing_pairs(self) -> Set[Tuple[int, int]]:
+        pairs = set()
+        if self.csv_path.exists():
```

Comment:
pour éviter d'avoir autant de blocs d'indentation
```suggestion
        if not self.csv_path.exists():
             return set()
```

Reply:
Corrected